_includes/tutorials/error-handling/kstreams/markup/dev/make-topology.adoc

```diff
@@ -0,0 +1,44 @@
+////
+In this file you describe the Kafka streams topology, and should cover the main points of the tutorial.
+The text assumes a method buildTopology exists and constructs the Kafka Streams application.  Feel free to modify the text below to suit your needs.
+////
+
+Here is the code we'll use to drive our tutorial
+
+[source, java]
+----
+
+   builder.stream(inputTopic, Consumed.with(Serdes.String(), Serdes.String()))
+                .mapValues(value -> {
+                    counter++;
+                    if (counter == 2 || counter == 8 || counter == 15) { <1>
+                        throw new IllegalStateException("It works on my box!!!");
+                    }
+                    return value.toUpperCase();
+                })
+                .to(outputTopic, Produced.with(Serdes.String(), Serdes.String()));
+
+    // Details left out for clarity
+    .......
+
+    // In the main method
+
+    final KafkaStreams streams = new KafkaStreams(topology, streamProps);
+    final MaxFailuresUncaughtExceptionHandler exceptionHandler = new MaxFailuresUncaughtExceptionHandler(maxFailures, maxTimeInterval); <2>
+    streams.setUncaughtExceptionHandler(exceptionHandler); <3>
+
+
+
+----
+
+<1> Simulating an error depending on the value of the counter
+<2> Instantiating the exception handler, the maxFailures (3) and maxTimeInterval (3600000 ms == 1 hour) variables get their values from the configuration files
+<3> Adding the handler to Kafka Streams
+
+From looking at the code here, we're injecting errors that occur based on the value of the counter.  We've also ensured that the number of errors meets the criteria for shutting down the application.
```

Comment:
For your consideration, move this sentence up to follow "Here is the code we’ll use to drive our tutorial"

```suggestion
This code ensures that the rate of errors (3 within a 1 hour window) meets the criteria for shutting down the application.
```